### PR TITLE
[SPARK-31010][SQL][FOLLOW-UP] Add Java UDF suggestion in error message of untyped Scala UDF

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Added Java UDF suggestion in the in error message of untyped Scala UDF.

### Why are the changes needed?

To help user migrate their use case from deprecate untyped Scala UDF to other supported UDF.

### Does this PR introduce any user-facing change?

No. It haven't been released.

### How was this patch tested?

Pass Jenkins.

### CODE REVIEW

**Comment by Ngone51** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -4864,9 +4864,10 @@ object functions {
         "information. Spark may blindly pass null to the Scala closure with primitive-type " +
         "argument, and the closure will see the default value of the Java type for the null " +
         "argument, e.g. `udf((x: Int) => x, IntegerType)`, the result is 0 for null input. " +
-        "You could use typed Scala UDF APIs (e.g. `udf((x: Int) => x)`) to avoid this problem, " +
-        s"or set ${SQLConf.LEGACY_ALLOW_UNTYPED_SCALA_UDF.key} to true and use this API with " +
-        s"caution."
+        "You could use typed Scala UDF APIs (e.g. `udf((x: Int) => x)`) or Java UDF (e.g. " +
+        "`udf(new UDF1[String, Int] { override def call(s: String): Int = s.length() }, " +
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: does Java UDF support scala `Int`?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -4864,9 +4864,10 @@ object functions {
...
+        "IntegerType)`) if input types are all non primitive to avoid this problem or set " +
+        "${SQLConf.LEGACY_ALLOW_UNTYPED_SCALA_UDF.key} to true and use this API with caution."
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: now there are 3 workarounds, can we list them nicely?
```
"To get rid of this error, you could:\n" + 
  "1. use typed Scala UDF, e.g. ...\n" +
  "2. use Java UDF, e.g. ..., if input types... \n" +
  "3. set .... "
```

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 3]
```diff
@@ -4864,9 +4864,12 @@ object functions {
         "information. Spark may blindly pass null to the Scala closure with primitive-type " +
         "argument, and the closure will see the default value of the Java type for the null " +
         "argument, e.g. `udf((x: Int) => x, IntegerType)`, the result is 0 for null input. " +
-        "You could use typed Scala UDF APIs (e.g. `udf((x: Int) => x)`) to avoid this problem, " +
-        s"or set ${SQLConf.LEGACY_ALLOW_UNTYPED_SCALA_UDF.key} to true and use this API with " +
-        s"caution."
+        "To get rid of this error, you could:\n" +
+        "1. use typed Scala UDF APIs, e.g. `udf((x: Int) => x)`\n" +
+        "2. use Java UDF APIs, e.g. `udf(new UDF1[String, Int] { override def call(s: String): " +
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: does Java UDF support scala `Int`?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 1 above)
Comment (Ngone51) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Yes.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 3 above)
Comment (Ngone51) [reply sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Yes, and I tested manually.

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by Ngone51** [comment]
thanks!

**Pull request closed without merging**